bin/generate-tickets.js

```diff
@@ -0,0 +1,143 @@
+}
+
+const getUnit = () => {
+    const result = { section: null, floor: null, unit: null }
+    const section = Data.property.sections[faker.datatype.number({ min: 0, max: Data.property.sections.length - 1 })]
+    result.section = section.name
+    const floor = section.floors[faker.datatype.number({ min: 0, max: section.floors.length - 1 })]
+    result.floor = floor.name
+    const unit = floor.units[faker.datatype.number({ min: 0, max: floor.units.length - 1 })]
+    result.unit = unit.label
+    return result
+}
+
+const getProblem = () => {
+    const start = Object.values(Data.classifiers.lvlOne)
+    return start[faker.datatype.number({ min: 0, max: start.length - 1 })]
+}
+
+const generateTicket = async (client, dayStamp) => {
+    const unit = getUnit()
+    const problem = getProblem()
+    const status = Data.statuses[faker.datatype.number({ min: 0, max: Data.statuses.length - 1 })]
+    const data = {
+        dv: 1,
+        sender: { dv: 1, fingerprint: 'import' },
+        clientName: `${faker.name.firstName()} ${faker.name.lastName()}`,
+        clientEmail: faker.internet.email(),
+        clientPhone: faker.phone.phoneNumber(),
+        details: faker.lorem.sentence(),
+        entranceName: unit.section,
+        floorName: unit.floor,
+        unitName: unit.unit,
+        source: { connect: { id: TICKET_OTHER_SOURCE_ID } },
+        classifier: { connect: { id: problem.id } },
+        // classifier: { connect: { id: problem.lvl1.id } },
+        operator: { connect: { id: Data.userId } },
+        assignee: { connect: { id: Data.userId } },
+        executor: { connect: { id: Data.userId } },
+        isEmergency: faker.datatype.boolean(),
+        isPaid: faker.datatype.boolean(),
+        organization: { connect: { id: Data.organizationId } },
+        property:  { connect: { id: Data.propertyId } },
+        status: { connect: { id: status.id } },
+    }
+    const result = await Ticket.create(client, data)
+    await setCreatedBy(result.id, dayStamp)
+}
+
+
+const ticketsToGeneratePerDay = [20, 50]
+
+const setCreatedBy = async (id, date) => {
+    await client.query(' Update "Ticket" SET "createdAt" = $1 WHERE id=$2 ', [
+        moment(date).utc().format('YYYY-MM-DD HH:mm:ss'),
+        id,
+    ])
+}
+
+const createTickets = async () => {
+    let counter = 0
+    const admin = await makeLoggedInAdminClient()
+    await createProperty(admin)
+    await loadUser(admin)
+    await loadModels(admin)
+    const dayStart = moment().utc().startOf('year')
+    const dayEnd = moment().utc()
+    let current = dayStart.add(6, 'hours')
+    let emergencyBreak = 10000
+    do {
+        const arr = Array(faker.datatype.number({ min: ticketsToGeneratePerDay[0], max: ticketsToGeneratePerDay[1] })).fill('')
+        for (const _ of arr) {
+            await generateTicket(admin, current.valueOf())
+        }
+        current = current.add(1, 'day')
+    } while (--emergencyBreak > 0 && current < dayEnd)
```

Comment:
Why do we need emergencyBreak check?

Reply:
@MrFoxPro This is to be sure that the disk does not run out of space. The script logic creates Tickets from the beginning of the year and there may be too many of them

Reply:
It's a good practice to add emergency breaks in while loops for js